Sources/NIO/SelectorGeneric.swift

```diff
@@ -0,0 +1,333 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2021 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIOConcurrencyHelpers
+
+internal enum SelectorLifecycleState {
+    case open
+    case closing
+    case closed
+}
+
+internal extension Optional {
+    func withUnsafeOptionalPointer<T>(_ body: (UnsafePointer<Wrapped>?) throws -> T) rethrows -> T {
+        if var this = self {
+            return try withUnsafePointer(to: &this) { x in
+                try body(x)
+            }
+        } else {
+            return try body(nil)
+        }
+    }
+}
+
+internal extension timespec {
```

Comment:
same here, `internal` to the `func`, or remove `internal` entirely.

Reply:
Fixed in e19e685.